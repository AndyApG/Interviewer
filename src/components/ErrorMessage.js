/* eslint-disable jsx-a11y/no-static-element-interactions */

import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { last } from 'lodash';
import { Button, Icon } from 'network-canvas-ui'; // eslint-disable-line
import { actionCreators as errorActions } from '../ducks/modules/errors';
import { Dialog } from '../components';

const getErrorMessage = error => !!error && error.message;

const getStack = error => !!error && error.stack;

/**
  * Renders a dialog box.
  */
const ErrorMessage = ({
  error,
  acknowledged,
  acknowledgeError,
}) => {
  const isActive = (!!error && !acknowledged);

  return (
    <Dialog
      title="Something went wrong!"
      show={isActive}
      type="error"
      hasCancelButton={false}
      confirmLabel="Acknowledged"
      onConfirm={acknowledgeError}
      additionalInformation={getStack(error)}
    >
      {error && ([
        <p>{getErrorMessage(error)}</p>,
        <p><strong>You can help us sending us the information below:</strong></p>,
      ])}
    </Dialog>
  );
};

ErrorMessage.propTypes = {
  acknowledged: PropTypes.bool,
  error: PropTypes.instanceOf(Error),
  acknowledgeError: PropTypes.func.isRequired,
};

ErrorMessage.defaultProps = {
  acknowledged: true,
  error: null,
};

const mapStateToProps = state => ({
  error: last(state.errors.errors) || null,
  acknowledged: state.errors.acknowledged,
});

const mapDispatchToProps = dispatch => ({
  acknowledgeError: bindActionCreators(errorActions.acknowledge, dispatch),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ErrorMessage);
