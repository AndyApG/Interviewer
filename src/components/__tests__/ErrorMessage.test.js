/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import { ErrorMessage } from '../ErrorMessage';

const mockProps = {
  errorMessage: 'There was an error',
  acknowledged: false,
  acknowledgeError: () => {},
};

describe('<ErrorMessage />', () => {
  it('renders ok', () => {
    const component = shallow(
      <ErrorMessage {...mockProps} />,
    );

    expect(component).toMatchSnapshot();
  });
});
