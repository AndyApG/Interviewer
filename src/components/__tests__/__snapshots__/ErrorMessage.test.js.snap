// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> renders ok 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorMessage
    acknowledgeError={[Function]}
    acknowledged={false}
    errorMessage="There was an error"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="dialog"
>
        <div
                className="dialog__background"
                transition-role="background"
        />
        <div
                className="dialog__window"
                onClick={[Function]}
                transition-role="window"
        >
                <div
                        className="dialog__main"
                >
                        <div
                                className="dialog__main-icon"
                        >
                                <Icon
                                        className=""
                                        color=""
                                        name="error"
                                        style={Object {}}
                                />
                        </div>
                        <div
                                className="dialog__main-content"
                        >
                                <h2
                                        className="dialog__main-title"
                                >
                                        Something went wrong!
                                </h2>
                                There was an error
                        </div>
                </div>
                <footer
                        className="dialog__footer"
                >
                        <Button
                                color=""
                                content="Acknowledged"
                                icon=""
                                iconPosition="right"
                                onClick={[Function]}
                                size=""
                        />
                </footer>
        </div>
</div>,
      "in": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="dialog__background"
            transition-role="background"
/>,
          <div
            className="dialog__window"
            onClick={[Function]}
            transition-role="window"
>
            <div
                        className="dialog__main"
            >
                        <div
                                    className="dialog__main-icon"
                        >
                                    <Icon
                                                className=""
                                                color=""
                                                name="error"
                                                style={Object {}}
                                    />
                        </div>
                        <div
                                    className="dialog__main-content"
                        >
                                    <h2
                                                className="dialog__main-title"
                                    >
                                                Something went wrong!
                                    </h2>
                                    There was an error
                        </div>
            </div>
            <footer
                        className="dialog__footer"
            >
                        <Button
                                    color=""
                                    content="Acknowledged"
                                    icon=""
                                    iconPosition="right"
                                    onClick={[Function]}
                                    size=""
                        />
            </footer>
</div>,
        ],
        "className": "dialog",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "dialog__background",
            "transition-role": "background",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="dialog__main"
>
                <div
                                className="dialog__main-icon"
                >
                                <Icon
                                                className=""
                                                color=""
                                                name="error"
                                                style={Object {}}
                                />
                </div>
                <div
                                className="dialog__main-content"
                >
                                <h2
                                                className="dialog__main-title"
                                >
                                                Something went wrong!
                                </h2>
                                There was an error
                </div>
</div>,
              <footer
                className="dialog__footer"
>
                <Button
                                color=""
                                content="Acknowledged"
                                icon=""
                                iconPosition="right"
                                onClick={[Function]}
                                size=""
                />
</footer>,
            ],
            "className": "dialog__window",
            "onClick": [Function],
            "transition-role": "window",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="dialog__main-icon"
>
                    <Icon
                                        className=""
                                        color=""
                                        name="error"
                                        style={Object {}}
                    />
</div>,
                  <div
                    className="dialog__main-content"
>
                    <h2
                                        className="dialog__main-title"
                    >
                                        Something went wrong!
                    </h2>
                    There was an error
</div>,
                ],
                "className": "dialog__main",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Icon
                      className=""
                      color=""
                      name="error"
                      style={Object {}}
/>,
                    "className": "dialog__main-icon",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "className": "",
                      "color": "",
                      "name": "error",
                      "style": Object {},
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h2
                        className="dialog__main-title"
>
                        Something went wrong!
</h2>,
                      "There was an error",
                    ],
                    "className": "dialog__main-content",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Something went wrong!",
                        "className": "dialog__main-title",
                      },
                      "ref": null,
                      "rendered": "Something went wrong!",
                      "type": "h2",
                    },
                    "There was an error",
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Button
                  color=""
                  content="Acknowledged"
                  icon=""
                  iconPosition="right"
                  onClick={[Function]}
                  size=""
/>,
                "className": "dialog__footer",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": null,
                  "color": "",
                  "content": "Acknowledged",
                  "icon": "",
                  "iconPosition": "right",
                  "onClick": [Function],
                  "size": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "footer",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="dialog"
>
          <div
                    className="dialog__background"
                    transition-role="background"
          />
          <div
                    className="dialog__window"
                    onClick={[Function]}
                    transition-role="window"
          >
                    <div
                              className="dialog__main"
                    >
                              <div
                                        className="dialog__main-icon"
                              >
                                        <Icon
                                                  className=""
                                                  color=""
                                                  name="error"
                                                  style={Object {}}
                                        />
                              </div>
                              <div
                                        className="dialog__main-content"
                              >
                                        <h2
                                                  className="dialog__main-title"
                                        >
                                                  Something went wrong!
                                        </h2>
                                        There was an error
                              </div>
                    </div>
                    <footer
                              className="dialog__footer"
                    >
                              <Button
                                        color=""
                                        content="Acknowledged"
                                        icon=""
                                        iconPosition="right"
                                        onClick={[Function]}
                                        size=""
                              />
                    </footer>
          </div>
</div>,
        "in": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="dialog__background"
              transition-role="background"
/>,
            <div
              className="dialog__window"
              onClick={[Function]}
              transition-role="window"
>
              <div
                            className="dialog__main"
              >
                            <div
                                          className="dialog__main-icon"
                            >
                                          <Icon
                                                        className=""
                                                        color=""
                                                        name="error"
                                                        style={Object {}}
                                          />
                            </div>
                            <div
                                          className="dialog__main-content"
                            >
                                          <h2
                                                        className="dialog__main-title"
                                          >
                                                        Something went wrong!
                                          </h2>
                                          There was an error
                            </div>
              </div>
              <footer
                            className="dialog__footer"
              >
                            <Button
                                          color=""
                                          content="Acknowledged"
                                          icon=""
                                          iconPosition="right"
                                          onClick={[Function]}
                                          size=""
                            />
              </footer>
</div>,
          ],
          "className": "dialog",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "dialog__background",
              "transition-role": "background",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="dialog__main"
>
                  <div
                                    className="dialog__main-icon"
                  >
                                    <Icon
                                                      className=""
                                                      color=""
                                                      name="error"
                                                      style={Object {}}
                                    />
                  </div>
                  <div
                                    className="dialog__main-content"
                  >
                                    <h2
                                                      className="dialog__main-title"
                                    >
                                                      Something went wrong!
                                    </h2>
                                    There was an error
                  </div>
</div>,
                <footer
                  className="dialog__footer"
>
                  <Button
                                    color=""
                                    content="Acknowledged"
                                    icon=""
                                    iconPosition="right"
                                    onClick={[Function]}
                                    size=""
                  />
</footer>,
              ],
              "className": "dialog__window",
              "onClick": [Function],
              "transition-role": "window",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="dialog__main-icon"
>
                      <Icon
                                            className=""
                                            color=""
                                            name="error"
                                            style={Object {}}
                      />
</div>,
                    <div
                      className="dialog__main-content"
>
                      <h2
                                            className="dialog__main-title"
                      >
                                            Something went wrong!
                      </h2>
                      There was an error
</div>,
                  ],
                  "className": "dialog__main",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Icon
                        className=""
                        color=""
                        name="error"
                        style={Object {}}
/>,
                      "className": "dialog__main-icon",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "className": "",
                        "color": "",
                        "name": "error",
                        "style": Object {},
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <h2
                          className="dialog__main-title"
>
                          Something went wrong!
</h2>,
                        "There was an error",
                      ],
                      "className": "dialog__main-content",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Something went wrong!",
                          "className": "dialog__main-title",
                        },
                        "ref": null,
                        "rendered": "Something went wrong!",
                        "type": "h2",
                      },
                      "There was an error",
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Button
                    color=""
                    content="Acknowledged"
                    icon=""
                    iconPosition="right"
                    onClick={[Function]}
                    size=""
/>,
                  "className": "dialog__footer",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": null,
                    "color": "",
                    "content": "Acknowledged",
                    "icon": "",
                    "iconPosition": "right",
                    "onClick": [Function],
                    "size": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "footer",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
